src/coreclr/vm/amd64/jithelpers_fastwritebarriers.S

```diff
@@ -415,6 +432,24 @@ PATCH_LABEL JIT_WriteBarrier_WriteWatch_PreGrow64_Patch_Label_Lower
         // Touch the card table entry, if not already dirty.
         shr     rdi, 0x0B
         NOP_2_BYTE // padding for alignment of constant
+
+#ifdef FEATURE_UNITY_NULLGC_WRITEBARRIER_PATCH
+
+PATCH_LABEL JIT_WriteBarrier_WriteWatch_PreGrow64_Patch_Label_Upper
+        movabs  r10, 0xF0F0F0F0F0F0F0F0
+
+        cmp     rsi, r10
+
+#ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
+        .byte 0x73, 0x3b
```

Comment:
The offset of 0x3b is probably wrong as there is an extra shr instruction on line 468.

Not sure what I am going to do about it as it may not be used.

Reply:
I think I have a better way to do this...

TLDR;
We need to force the bReqUpperBoundsCheck bool to be true here...

https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/amd64/jitinterfaceamd64.cpp#L833

Will swap this out and try again.